### DISCUSSION TITLE
Linking to docs.rs/<crate>/ when crate version is latest

### DISCUSSION THREAD

**Discussion by jsha** [rust-lang/crates.io#4290] [Feature Requests]
In https://github.com/rust-lang/docs.rs/issues/1438 I'm working on improving ability for search engines to reliably find the latest version of a crate. The main part of the docs.rs work is done: https://docs.rs/rustls/ now redirects to https://docs.rs/rustls/latest/rustls/, not https://docs.rs/rustls/0.20.2/rustls/.

It would be helpful if crates.io also linked to the latest version. In particular, I'm proposing: If a user is viewing the latest version of crate, the documentation link on crates.io should point to `https://docs.rs/<cratename>/` instead of a specific versioned URL. If the user is viewing an older version, the documentation link on crates.io should continue to point to a versioned URL as it does today.

**Comment by Turbo87** [comment]
hmm, I think that would only work if both systems used the exact same algorithm to determine what the latest version is 🤔

**Reply by syphar** [reply]
> hmm, I think that would only work if both systems used the exact same algorithm to determine what the latest version is 🤔

How does crates.io do this? Could you point us to the place in the code? 

In docs.rs it does _not_ depend on the build-status: 

https://github.com/rust-lang/docs.rs/blob/6e354ec15c6652660afd8a071b3d09b3b8eb6b68/src/web/crate_details.rs#L229-L234

**Reply by Turbo87** [reply]
https://github.com/rust-lang/crates.io/blob/56301b61af41acb28067458e2f2597be0365249c/app/models/crate.js#L30-L43 and https://github.com/rust-lang/crates.io/blob/56301b61af41acb28067458e2f2597be0365249c/app/routes/crate/version.js#L30-L31 are the relevant code locations, but that's not necessarily stable.

I've been thinking about exposing something like `default_version` on the API which would be calculated by the backend. We are currently exposing `max_stable_version` and `max_version`, but both of them don't take yanked crates into account and it puts the responsibility on the API client to figure out what the default should be.

**Reply by syphar** [reply]
Hm... on top of that:

Depending on the build-prioritization it can sometimes take 12 hours for the build to happen. 

So the only way that `/latest/` always points to the correct, exact version would be docs.rs reporting back build-status to crates.io.
